### PR TITLE
Update RandomAmr to actually do random refinement

### PR DESCRIPTION
**Pull request by kidder**
Previously no refinement actions were being run; only the reduction done by CheckDomain was being tested.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: tests/InputFiles/ExampleExecutables/RandomAmr1D.yaml [hunk 1]
```diff
@@ -3,10 +3,18 @@
...
 Testing:
-  Check: parse;execute
+  Check: parse;execute_check_output
+ExpectedOutput:
+ - SpectreCheckpoint000000
+Timeout: 10
 
 ---
 
+Criteria:
```
Comment (knelli2) [inline tests/InputFiles/ExampleExecutables/RandomAmr1D.yaml:13]: Not for this PR, but now that I see this in an input file, I think the yaml group should be called `AmrCriteria`. I think that would be clearer.

File: src/Executables/Examples/RandomAmr/RandomAmr.hpp [hunk 2]
```diff
@@ -73,20 +96,44 @@ struct RandomAmrMetavars {
...
   using component_list =
-      tmpl::list<amr::Component<RandomAmrMetavars>, amr_element_component>;
+      tmpl::list<amr::Component<RandomAmrMetavars>, dg_element_array>;
 
   // NOLINTNEXTLINE(google-runtime-references)
   void pup(PUP::er& /*p*/) {}
 };
 
+template <typename Component>
+void register_amr_callbacks() {
```
Comment (nilsvu) [inline src/Executables/Examples/RandomAmr/RandomAmr.hpp:115]: Do you have plans to move this to a reusable place?

File: tests/InputFiles/ExampleExecutables/RandomAmr1D.yaml (hunk 1 above)
Comment (nilsvu) [reply tests/InputFiles/ExampleExecutables/RandomAmr1D.yaml:13]: I think just add an `Amr` option group and put this tag in there.

**Review by nilsvu — APPROVED** [review]

File: tests/InputFiles/ExampleExecutables/RandomAmr1D.yaml (hunk 1 above)
Comment (kidder) [reply tests/InputFiles/ExampleExecutables/RandomAmr1D.yaml:13]: okay, I'll do that in my next AMR PR

File: src/Executables/Examples/RandomAmr/RandomAmr.hpp (hunk 2 above)
Comment (kidder) [reply src/Executables/Examples/RandomAmr/RandomAmr.hpp:115]: Yes, as I don't want to copy it into all the executables...  I'll do this in my next AMR PR as well

**Pull request merged**